src/VisualStudio/IntegrationTest/TestUtilities/TemporaryTextFile.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities
+{
+    /// <summary>
+    /// Provides methods for supporting temporary files within Roslyn tests.
+    /// </summary>
+    public class TemporaryTextFile : IDisposable
+    {
+        private string _fileName;
+        private string _content;
+        private string _path;
+
+        public TemporaryTextFile(string fileName, string content)
+        {
+            _fileName = fileName;
+            _content = content;
+            _path = Path.Combine(Path.GetTempPath(), "RoslynTests", "TemporaryTextFile");
```

Comment:
The existing logic we have is here: http://source.roslyn.io/#Microsoft.VisualStudio.IntegrationTest.Utilities/InProcess/SolutionExplorer_InProc.cs,91,

Basically we just do:
```
var path = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
IntegrationHelper.DeleteDirectoryRecursively(path);
```

I recommended that this code be moved to the Helper class and called from both locations (that way the logic is centralized).